src/stan/callbacks/json_writer.hpp

```diff
@@ -265,8 +268,8 @@ class json_writer {
       return;
     write_record_comma_if_needed();
     *output_ << "{";
-    record_needs_comma_ = false;
     record_depth_++;
+    record_element_needs_comma_ = false;
```

Comment:
Why was this changed from `record_needs_comma` to `record_element_needs_comma_`?

Reply:
because that logic has been moved into `write_record_comma_if_needed`

Reply:
added more tests to make sure this works properly.